src/test/integration/reduxPersist.js

```diff
@@ -0,0 +1,122 @@
+import { createStore } from 'redux';
+
+import { persistCombineReducers, persistStore } from 'redux-persist';
+import storage from 'redux-persist/lib/storage';
+
+import {
+    ORM, Session, Model as OrmModel, createReducer, createSelector
+} from '../..';
+
+import { createTestModels } from '../helpers';
+import { STATE_FLAG } from '../../constants';
+
+describe('Redux Persist integration', () => {
+    let orm;
+    let Book;
+    let Cover;
+    let Genre;
+    let Tag;
+    let Author;
+    let Publisher;
+    let Movie;
+    let emptyState;
+    let ormState;
+
+    let reducer;
+    let rootReducer;
+    const stateSelector = state => state.orm;
+
+    let store;
+
+    const CREATE_MOVIE = 'CREATE_MOVIE';
+    const UPSERT_MOVIE = 'UPSERT_MOVIE';
+    const CREATE_PUBLISHER = 'CREATE_PUBLISHER';
+    const CREATE_CUSTOM_MODEL = 'CREATE_CUSTOM_MODEL';
+
+    const createModelReducers = () => {
+        Author.reducer = jest.fn();
+        Book.reducer = jest.fn();
+        Cover.reducer = jest.fn();
+        Genre.reducer = jest.fn();
+        Tag.reducer = jest.fn();
+        Movie.reducer = jest.fn((action, Model, _session) => {
+            switch (action.type) {
+            case CREATE_MOVIE:
+                Model.create(action.payload);
+                break;
+            case UPSERT_MOVIE:
+                Model.upsert(action.payload);
+                break;
+            default: break;
+            }
+        });
+        Publisher.reducer = jest.fn((action, Model, _session) => {
+            switch (action.type) {
+            case CREATE_PUBLISHER:
+                Model.create(action.payload);
+                break;
+            default: break;
+            }
+        });
+    };
+
+    beforeEach(() => {
+        ({
+            Book,
+            Cover,
+            Genre,
+            Tag,
+            Author,
+            Movie,
+            Publisher,
+        } = createTestModels());
+        orm = new ORM({ stateSelector });
+        orm.register(Book, Cover, Genre, Tag, Author, Movie, Publisher);
+        createModelReducers();
+        emptyState = orm.getEmptyState();
+        ormState = emptyState;
+        reducer = createReducer(orm);
+        const persistConfig = {
```

Comment:
I got it working (failing?) after some fiddling. Because the default merge strategy is a shallow merge one level deep, redux-persist doesn't overwrite the initial `STATE_FLAG`.
https://github.com/rt2zz/redux-persist#state-reconciler

Add 
```
import hardSet from "redux-persist/lib/stateReconciler/hardSet";
...
        const persistConfig = {
            key: "root",
            stateReconciler: hardSet,
            storage,
            whitelist: ["orm"],
        };
```
That way it overwrites the whole store with the persisted state. I also needed to do some additional messing with jest to get it to wait for all of the promises to resolve.